arch/arm/pl011.c

```diff
@@ -0,0 +1,71 @@
+/**
+ * @file arch/arm/pl011.c
+ *
+ * ARM PrimeCell UART PL011.
+ */
+#include <arch/pl011.h>
+#include <arch/barrier.h>
+
+static inline uint32_t pl011_read(const volatile void *addr)
+{
+    uint32_t val;
+#ifdef CONFIG_ARM_64
+    asm volatile("ldr %w0, [%1]\n" : "=r" (val) : "r" (addr));
```

Comment:
How about introducing `ioread` and `iowrite`? So other driver in the future can use it without ifdefing.

Reply:
Definitely it would be better. I will introduce them.